### PR TITLE
Add simple support for package:pageloader

### PR DESCRIPTION
**Pull request by matanlurey**
Branched from https://github.com/dart-lang/angular_test/pull/10
Closes https://github.com/dart-lang/angular_test/issues/11

### CODE REVIEW

File: lib/src/frontend/fixture.dart [hunk 1]
```diff
@@ -50,6 +53,22 @@ class NgTestFixture<T> {
...
+    await update();
+    return _pageLoader.getInstance/*<T>*/(pageObjectType);
+  }
+
+  /// Returns a page loader instance represnting this test fixture.
+  PageLoader get _pageLoader => _pageLoaderInstance ??= new HtmlPageLoader(
+        rootElement,
+        executeSyncedFn: (fn) async {
+          await fn();
+          return () => update();
```
Comment (natebosch) [inline lib/src/frontend/fixture.dart]: [nit] `return update;` since `update` won't change

File: lib/src/frontend/fixture.dart [hunk 2]
```diff
@@ -50,6 +53,22 @@ class NgTestFixture<T> {
     activeTest = null;
   }
 
+  /// Return a page object representing [pageObjectType] from the DOM.
+  Future/*<T>*/ getPageObject/*<T>*/(Type pageObjectType) async {
+    await update();
+    return _pageLoader.getInstance/*<T>*/(pageObjectType);
+  }
+
+  /// Returns a page loader instance represnting this test fixture.
```
Comment (natebosch) [inline lib/src/frontend/fixture.dart]: Prefer a noun phrase for 'get' methods.

also [nit] typo in representing.

File: test/frontend/pageloader_test.dart [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+@Tags(const ['aot'])
+@TestOn('browser')
+import 'package:angular2/angular2.dart';
+import 'package:angular_test/angular_test.dart';
+import 'package:pageloader/objects.dart';
+import 'package:test/test.dart';
+
+@AngularEntrypoint()
+void main() {
+  tearDown(() => disposeAnyRunningTest());
```
Comment (natebosch) [inline test/frontend/pageloader_test.dart]: `disposeAnyRunningTest` is not a reference that can change

`tearDown(disposeAnyRunningTest);`

File: lib/src/frontend/fixture.dart [hunk 4]
```diff
@@ -50,6 +53,22 @@ class NgTestFixture<T> {
     activeTest = null;
   }
 
+  /// Return a page object representing [pageObjectType] from the DOM.
+  Future/*<T>*/ getPageObject/*<T>*/(Type pageObjectType) async {
```
Comment (natebosch) [inline lib/src/frontend/fixture.dart]: how about something like `findPageObject`?

`get` adds zero information to a method name - is there any more descriptive word we can use?

Comment (matanlurey) [reply lib/src/frontend/fixture.dart]: I guess we could say `resolve` - probably more accurate than `find`. PTAL.

File: lib/src/frontend/fixture.dart (hunk 2 above)
Comment (matanlurey) [reply lib/src/frontend/fixture.dart]: Done.

File: lib/src/frontend/fixture.dart (hunk 1 above)
Comment (matanlurey) [reply lib/src/frontend/fixture.dart]: Done.

File: test/frontend/pageloader_test.dart (hunk 3 above)
Comment (matanlurey) [reply test/frontend/pageloader_test.dart]: Done.

**Review by natebosch — APPROVED** [review]
LGTM

**Comment by matanlurey** [comment]
Waiting for travis to make sure I fixed it correctly.

**Pull request merged**